client/mapView/MapView.cpp

```diff
@@ -115,9 +124,49 @@ void MapView::onMapScrolled(const Point & distance)
 
 void MapView::onMapSwiped(const Point & viewPosition)
 {
+	if(settings["adventure"]["smoothDragging"].Bool())
+		swipeHistory.push_back(std::pair<uint32_t, Point>(GH.input().getTicks(), viewPosition));
+
 	controller->setViewCenter(model->getMapViewCenter() + viewPosition, model->getLevel());
 }
 
+void MapView::postSwipe(uint32_t msPassed) {
+	if(!actions->dragActive)
+	{
+		if(swipeHistory.size() > 1)
+		{
+			Point diff = Point(0, 0);
+			std::pair<uint32_t, Point> firstAccepted;
+			uint32_t now = GH.input().getTicks();
+			for (auto & x : swipeHistory) {
+				if(now - x.first < 150) { // only the last 150 ms are catched
+					if(firstAccepted.first == 0)
+						firstAccepted = x;
+					diff += x.second;

```

Comment:
Do we need to accumulate?
Why can't we compare firstAccepted.second and swipeHistory.back().second?

Reply:
Yes it's needed. Because each entry in `swipeHistory` is relative to the entry before. So, if we want to get the correct speed over multiple entries (time range), we have to sum up.